### PR TITLE
Added basic testing that streamr-ganache started correctly, also added features

### PR DESCRIPTION
**Pull request by jtakalai**
now passes even though EE wasn't running, see:
// TODO: test that some E&E test-products are in the marketplace

also not hooked up the smoke test to deploy_docker.sh yet

Feature added: 10000 test accounts that have 100 ETH, for integration-tests, so that each can have their own wallet (no nonce problems if tests are run in parallel)

### CODE REVIEW

**Review by mikhaelsantos — COMMENTED** [review]
The semantics are confusing in this PR.

The word "_Test_" is used to verify a piece of software worked.
The word "_Smoke_ Test" is in our context used to verify if the container is working.

What you have is  "_Integration test_" that is verifying that your contract deployment code worked with ganache.

The _smoke test_ in our context is for verifying that a built container is working.

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@ env:
   - secure: Qi+QPGx8olG4Xu5xa4LZPP1yMTiRtyEbXgW8Tr2XBPRQO7DL8wjmC0wgOig0+UdmtMBAoS9QC9qnLl06/Ry6Bppsdg1kdNp71kQLU7BCDz6aJekpatpoUJ2C4KGhNXVcBEX3v7rH1zlUVwDfcrjD2tgH4lRINOd7O8THuo6fA9XiRAXJHImukXtbcoRHjqBZSq7wAHL9IsdR9GaRTx40P6sXoJeRaP/PBw+diZkAO8WSDxaVgdINVyNXRiCfXiGppp4ql1C4tuFfpnERsbEUGrQ+vttOS5x8Xj/5MDpuYts1hKUfxX+dA7C/LYj131/2c0eKvCfNGKhzO/aKV5Ucf5k2+0wWOMAPrH788d3nvthqNY2UJOlq/Qqh7NXQBVy0BEcDCGGzjYXv39vxNJ2YUaJTmLGpwo3FXHDNSm7FXol9oN31Sp0vkT78zgvBemplbB4KsJ1jmWzvJGatqjbP3POCrU6jM7wLg3EsRDLeKyJf+oxgd63ZMD3wCKXAjaKs/0EkTz4R0a8aIBzdC/rDIAQHORy/92Cuta7UGPCHIqUSvrmWr7yf8y97rygmG2OB41yx5eVL7mPnt2e6oxEyRdENM3D7lYpGfynwprT+p9f3pBxtpaztL/E52JvGg1hlTzC/Ng+EZDO85iUegzzFYzCSJkivoBHJIuyUpvib4vA=
 jobs:
   include:
+  - stage: test
```
Comment (mikhaelsantos) [inline .travis.yml]: Poor naming for stage

Comment (mikhaelsantos) [inline .travis.yml:19]: Poor naming for stage

File: .travis.yml [hunk 2]
```diff
@@ -16,6 +16,11 @@ env:
...
+    script:
+    - node ./index.js &
+    - sleep 2m
+    - node ./smoke_test.js
```
Comment (mikhaelsantos) [inline .travis.yml:23]: Confusion between Tests (Unit/Integration) with, what we are calling smoke tests, that verifies if built container has started.

File: smoke_test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    for (const wallet of wallets.slice(1, 10)) {
+        const ethBalance = await wallet.getBalance()
+        assert.strictEqual(ethBalance.toString(), "100000000000000000000")
+
+        const token = new Contract(tokenAddress, TokenJson.abi, wallet)
+        const tokenBalance = await token.balanceOf(wallet.address)
+        assert.strictEqual(tokenBalance.toString(), "1000000000000000000000000")
+    }
+
+    /* the following snippet is from index.js, TODO: make it work here
```
Comment (mikhaelsantos) [inline smoke_test.js:52]: Remove comments. Create Jira ticket for task, or this to do will be forgotten.
At maximum add TODO: JIRA-Ticket

File: smoke_test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env node
```
Comment (mikhaelsantos) [inline smoke_test.js:1]: This whole file is not we what we are classifying as a smoke test check here: https://github.com/streamr-dev/streamr-infra/wiki/Docker-Smoke-Test#smoke-test

**Review by mikhaelsantos — CHANGES_REQUESTED** [review]
More information about smoke tests here:
https://github.com/streamr-dev/streamr-infra/wiki/Docker-Smoke-Test and in the docker circle channel.

**Comment by jtakalai** [comment]
ok well maybe the PR could be re-organized later

**Pull request closed without merging**